.github/workflows/rust-mirai.yml

```diff
@@ -0,0 +1,64 @@
+name: Rust
+
+on:
+  pull_request:
+    types: ['opened', 'edited', 'reopened', 'synchronize']
+
+jobs:
+  install_and_run_MIRAI:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Install Z3
+      uses: pavpanchekha/setup-z3@v1.2
+    - name: Checkout MIRAI
+      uses: actions/checkout@v2
+      with:
+        repository: 'facebookexperimental/MIRAI'
+        path: 'MIRAI'
+    - name: Install MIRAI
+      working-directory: ./MIRAI
+      run: |
+        rustup component add rustc-dev
+        RUSTFLAGS='-Clink-arg=-L./binaries -Clink-arg=-lstdc++' cargo install --path ./checker
+    - name: Checkout libra
+      uses: actions/checkout@v2
+      with:
+        path: 'libra'
+    - name: Build libra master then checkout PR
+      working-directory: ./libra
+      run: |
+        rustup override set `cat ../MIRAI/rust-toolchain`
+        rustup component add rustc-dev
+        git fetch origin master
+        git checkout master
+        RUSTFLAGS="-Z always_encode_mir" cargo build
+        git checkout -
+    - name: Run MIRAI on PR
+      working-directory: ./libra
+      run: |
+        rustup show
+        RUSTC_WRAPPER=mirai RUSTFLAGS="-Z always_encode_mir" cargo build > ../mirai_results
+        cat ../mirai_results
+    - name: Post comment with MIRAI warnings
+      if: success()
+      uses: actions/github-script@0.4.0
+      with:
+        github-token: ${{secrets.GITHUB_TOKEN}}
```

Comment:
It needs to setup before we land this PR.  Alternatively, what's the down side of having the failed job reporting the status back to the PR?

Reply:
that's already setup by default, it makes [github-actions](https://github.com/mimoo/libra/pull/3/#issuecomment-577440330) comment. We might want to brand MIRAI with our own bot though :p